solutions/ac-GCDOnBlackboard.mdx

```diff
@@ -114,4 +114,64 @@ print(ans)
 ```
 
 </PySection>
+
+	
+<JavaSection>
+	
+```java
+import java.util.*;
+import java.io.*;
+
+public class GCDBoard {
+	static final int MAX_N = (int) 1e5 + 5;
+
+	public static void main(String[] args) throws IOException {
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		int N = Integer.parseInt(br.readLine());
+		StringTokenizer st = new StringTokenizer(br.readLine());
+
+		int[] arr = new int[MAX_N];
```

Comment:
```suggestion
		int[] arr = new int[n];
```

Reply:
do this or put the arrays outside of the main method.

Reply:
use N, but the renaming isn't needed

Reply:
Ummm, if you use n then the code won't work......

Reply:
because the for loop goes from 1 to N. Should I keep int[] arr = new int[MAX_N] or int[] arr = new int[n+1]?

Reply:
n + 1